packages/data-stores/src/site/utils.ts

```diff
@@ -13,13 +14,17 @@ export const createCustomHomeTemplateContent = (
 	}
 
 	if ( hasSections ) {
-		// blockGap":"0" removes the theme blockGap from the main group while allowing users to change it from the editor
-		content.push( `
+		if ( wrapMainHtmlInMainGroup ) {
+			// blockGap":"0" removes the theme blockGap from the main group while allowing users to change it from the editor
+			content.push( `
 <!-- wp:group {"tagName":"main","style":{"spacing":{"blockGap":"0"}}} -->
```

Comment:
I wonder if we actually need this wrapper if there is HTML code.

Seems it was added in the early days of the Pattern Assembler in order to avoid creating an empty template (see https://github.com/Automattic/wp-calypso/pull/69241).

Maybe we can do something like this?

```js
if ( mainHtml ) {
	content.push( mainHtml );
} else {
	content.push( `
<!-- wp:group {"tagName":"main","style":{"spacing":{"blockGap":"0"}}} -->
	<main class="wp-block-group"></main>
<!-- /wp:group -->
	` );
```

cc @Automattic/lego

Reply:
No idea. Let's see what Lego thinks :)

Reply:
I am Lego but I totally forgot why 😂 

My best guess is that we want to replicate Creatio 2 / Assembler's original index/home template, which always wraps the main patterns inside that wrapper.

Reply:
I remember spmething about avoiding the core template modal when landing in the editor without patterns selected.

Reply:
But I also remember we added an empty paragraph to fix that so it's not that.

Reply:
@fushar @miksansegundo, soo... are you fine with adding the argument flag? :D

Reply:
Sorry, which argument flag?

I remember now that group with blockGap 0 is for this [issue](https://github.com/Automattic/wp-calypso/pull/80212). If that's not reproducible then we can remove it.

Reply:
Considering we're not confident about why the main group was added, let's maybe play safe for now and keep the current behavior enabled by default.

> Sorry, which argument flag?

The new `wrapMainHtmlInMainGroup` param.

Reply:
Hmm. It seems that the issue is reproducing in Assembler too. It's probably a change in Gutenberg that re-introduced the original problem.

I'll abandon this PR and open another one that fixes the issue for both the Assembler and RMTs! Thanks @miksansegundo! 🙇